samcli/local/docker/container.py

```diff
@@ -272,6 +274,31 @@ def start(self, input_data=None):
         # Start the container
         real_container.start()
 
+        # Wait for port to be open
+        self.wait_for_port()
+
+    def wait_for_port(self):
+        """
+        Waits until the host machine port that Docker binds to is open.
+        """
+        start_time = time.time()
+        sleep = 0.1
+        timeout = 10
+        while True:
+            a_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+            location = (self._container_host_interface, self.rapid_port_host)
+            is_port_open = not a_socket.connect_ex(location)
+            a_socket.close()
```

Comment:
https://www.kite.com/python/answers/how-to-check-if-a-network-port-is-open-in-python

Should I include this as a comment in code?

Reply:
I think it is Ok

Reply:
I think a comment here just explaining what's going on and how this is waiting for a non-error socket connection to the container could be helpful.

Reply:
added a small comment